presto-main/src/main/java/io/prestosql/memory/MemoryManagerConfig.java

```diff
@@ -95,10 +98,25 @@ public MemoryManagerConfig setMaxQueryTotalMemory(DataSize maxQueryTotalMemory)
         return this;
     }
 
-    public static final class LowMemoryKillerPolicy
+    public enum LowMemoryKillerPolicy
     {
-        public static final String NONE = "none";
-        public static final String TOTAL_RESERVATION = "total-reservation";
-        public static final String TOTAL_RESERVATION_ON_BLOCKED_NODES = "total-reservation-on-blocked-nodes";
+        NONE,
+        TOTAL_RESERVATION,
+        TOTAL_RESERVATION_ON_BLOCKED_NODES,
+        /**/;
+
+        public static LowMemoryKillerPolicy fromString(String value)
```

Comment:
This doesn't work because I unconditionally return here:
https://github.com/airlift/airlift/blob/6bdf92c96e772da7200c0e3b5771b57eeb4edd8e/configuration/src/main/java/io/airlift/configuration/ConfigurationFactory.java#L604
I overlooked the fact an enum may define `fromString` conversion.

- for an enum without `fromString` method, it makes sense to return
- for an enum with `fromString` method
  - currently `fromString` method cannot be used (regression in airlift)
  - if we don't return on "not found", two sets of values will be accepted: case-insensitive-by-name and whatever the method accepts

Maybe we should not attempt to `valueOf` or case-insensitive-by-name when `fromString` is defined?

Reply:
Good find, @findepi . 

I wanted to share that I have a WIP patch to do something similar for `ExecutionPolicy`. There is an additional concern there. Since it is also a session property, we need to make sure that the decoder does something more specific than `return Enum.valueOf(type, ((String) value).toUpperCase(ENGLISH))`. Both these cases (configuration and session property decoding) can use the same enum method defined in airlift. What do you think?

Reply:
@phd3 interesting idea. Not sure how clean/hack-ish it would need to be in practice. Would you like to try out and see?

(also related: https://github.com/airlift/airlift/pull/759)